code/control/Dispatcher.php

```diff
@@ -50,15 +64,22 @@ public function asJSONFormFieldErrors($code, $fieldErrors) {
 		]));
 		$response->setStatusCode($code);
 		return $response;
-
 	}
 
-	public function asJSON($data) {
+	public function asJSON($data = []) {
 		$securityToken = $this->getSecurityToken();
 		$securityToken->reset();
 		$data['NewSecurityID'] = $securityToken->getValue();
 
 		$response = $this->getResponse();
+
+		//Mask redirects in 200 responses, because JS never has an option not to follow redirects -
```

Comment:
Changed the comment, hopefully more readable now? The whole thing really needs a sprinkle of comments.